.buildkite/pipeline.yml

```diff
@@ -33,32 +33,76 @@ x-common-params:
     build.branch != 'trunk' && build.branch !~ /^release.*/ && build.branch !~ /^dependabot\/submodules.*/
 
 steps:
-  - label: Lint and Unit Tests
-    key: unit-tests
+  - label: Lint
+    key: lint
     plugins:
       - *gb-mobile-docker-container
+      - *ci_toolkit_plugin
+      - *nvm_plugin
+      - *git-cache-plugin
     command: |
+      echo "--- :docker: Additional Docker image setup"
       source /root/.bashrc
 
-      echo "--- :node: Setup Node environment"
+      echo "--- :docker::node: Setup Node environment"
       nvm install && nvm use
```

Comment:
I tried relying on the [nvm plugin](https://github.com/Automattic/nvm-buildkite-plugin) but, despite successfully installing nvm and the node version form the `.nvmrc` file in the repo, when the commands run in Docker, the Node version was not set as expected:

![image](https://github.com/wordpress-mobile/gutenberg-mobile/assets/1218433/0cd183bc-ccf0-4900-8cf5-948c98c1b708)

See https://buildkite.com/automattic/gutenberg-mobile/builds/7569#018b40c7-f76a-4ace-ae1c-2b73f44c5ab5

Maybe that has got to do with the Docker container spinning up after nvm did its setup?

Reply:
> but my perception of Docker is that its containers are generally used to create an environment with all the necessary requirement ready to go.

That's my understanding of how Docker should be used, too.

The image we have does setup `nvm` and installs the LTS version, but it maybe there's more to do to enable automatically switching to the version from `.nvmrc` when changing the folder?

https://github.com/wordpress-mobile/docker-gb-mobile-image/blob/25f008b93f9fe190f76c68f3a9719b45ad99a6cc/Dockerfile#L16

My approach at the moment is to iterate locally, which is faster. Once an implementation is locked in, we can push the additional setup in the Docker image (if we'll still be using it) to improve performance. What do you think?

Reply:
> The image we have does setup `nvm` and installs the LTS version, but it maybe there's more to do to enable automatically switching to the version from `.nvmrc` when changing the folder?

Yes, this may be the issue. `nvm` does not automatically switch Node.js versions when changing directories. [Custom scripting](https://github.com/nvm-sh/nvm/tree/v0.39.5#deeper-shell-integration) is required to enable such a feature with `nvm`. 

It appears the nvm plugin does call `nvm use` — which should be enough to ensure the correct Node.js version is set for subsequent commands — but maybe later commands are executed in a different session? 

> My approach at the moment is to iterate locally, which is faster. Once an implementation is locked in, we can push the additional setup in the Docker image (if we'll still be using it) to improve performance. What do you think?

This makes a lot of sense. 👍🏻